src/tags/for_block.rs

```diff
@@ -394,16 +386,42 @@ impl Renderable for TableRow {
         let range = iter_slice(&mut range, limit, offset, false);
 
         context.run_in_scope(|mut scope| {
-            let mut row_num = 0;
-            let mut col_num = 0;
+            let mut helper_vars = Object::new();
+            helper_vars.insert("length".into(), Value::scalar(range.len() as i32));
+
             for (i, v) in range.iter().enumerate() {
-                if position_in_row(i, cols) == 0 {
-                    row_num += 1;
-                    col_num = 0;
-                    write!(writer, "<tr class=\"row{}\">", row_num).chain("Failed to render")?;
+                let (col_index, row_index) = match cols {
+                    Some(cols) => (i % cols, i / cols),
+                    None => (i, 0),
+                };
+
+                let first = i == 0;
+                let last = i == (range.len() - 1);
+                let col_first = col_index == 0;
+                let col_last = cols.filter(|&cols| col_index + 1 == cols).is_some() || last;
+
+                if col_first {
+                    write!(writer, "<tr class=\"row{}\">", row_index + 1)
+                        .chain("Failed to render")?;
                 }
-                col_num += 1;
-                write!(writer, "<td class=\"col{}\">", col_num).chain("Failed to render")?;
+                write!(writer, "<td class=\"col{}\">", col_index + 1).chain("Failed to render")?;
```

Comment:
I think I'd prefer `helper_vars` to be initialized first.  That leaves it so we have all the writes in a row, making it easier to see whats happening.

If for-loop isn't doing this, then oops (granted, I didn't write it)